Source: From Void Linux
Reason: Fix build with musl libc
Upstream: no

--- a/mtp/backend/posix/Exception.cpp	2017-06-23 12:08:10.000000000 +0200
+++ b/mtp/backend/posix/Exception.cpp	2018-03-25 17:53:15.359371456 +0200
@@ -35,7 +35,7 @@
 	std::string Exception::GetErrorMessage(int returnCode)
 	{
 		char buf[1024];
-#ifdef _GNU_SOURCE
+#if defined(_GNU_SOURCE) && defined(__GLIBC__)
 		std::string text(strerror_r(returnCode, buf, sizeof(buf)));
 #else
 		int r = strerror_r(returnCode, buf, sizeof(buf));
@@ -44,4 +44,4 @@
 		return text;
 	}
 
-}}
\ Kein Zeilenumbruch am Dateiende.
+}}
--- a/fuse/Exception.h	2017-06-23 12:08:10.000000000 +0200
+++ b/fuse/Exception.h	2018-03-25 17:57:06.478366801 +0200
@@ -36,7 +36,7 @@
 		static std::string GetErrorMessage(int returnCode)
 		{
 			char buf[1024];
-#ifdef _GNU_SOURCE
+#if defined(_GNU_SOURCE) && defined(__GLIBC__)
 			std::string text(strerror_r(returnCode, buf, sizeof(buf)));
 #else
 			int r = strerror_r(returnCode, buf, sizeof(buf));
